src/BaseDAO/ShareTest/Common.hs

```diff
@@ -48,10 +48,30 @@ type IsLorentz = Bool
 
 -- | Used for check error from ligo contract
 expectFailed
- :: forall caps base m
-  . (MonadNettest caps base m)
-  => Address -> MText -> m () -> m ()
-expectFailed addr val = flip expectFailure (NettestFailedWith addr val)
+ :: forall caps base m a
+  . (MonadNettest caps base m, HasCallStack)
+  => Address -> MText -> m a -> m ()
+expectFailed addr val = withFrozenCallStack $
```

Comment:
I may be wrong, but it seems to me that everywhere this is called the first argument is always made with `toAddress`, since this is an helper function to reduce duplication, why not put `toAddress` in it an remove it from the caller?